src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -163,89 +155,125 @@ SelectClause Visitor::visitTypesafe(Parser::SelectClauseContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-VarOrAlias Visitor::visitTypesafe(Parser::VarOrAliasContext* ctx) {
+VarOrAlias Visitor::visit(Parser::VarOrAliasContext* ctx) {
   return visitAlternative<VarOrAlias>(ctx->var(), ctx->alias());
 }
 
 // ____________________________________________________________________________________
-Alias Visitor::visitTypesafe(Parser::AliasContext* ctx) {
+Alias Visitor::visit(Parser::AliasContext* ctx) {
   // A SPARQL alias has only one child, namely the contents within
   // parentheses.
-  return visitTypesafe(ctx->aliasWithoutBrackets());
+  return visit(ctx->aliasWithoutBrackets());
 }
 
 // ____________________________________________________________________________________
-Alias Visitor::visitTypesafe(Parser::AliasWithoutBracketsContext* ctx) {
-  return {{visitTypesafe(ctx->expression())}, visitTypesafe(ctx->var())};
+Alias Visitor::visit(Parser::AliasWithoutBracketsContext* ctx) {
+  return {{visit(ctx->expression())}, visit(ctx->var())};
 }
 
 // ____________________________________________________________________________________
-ParsedQuery Visitor::visitTypesafe(Parser::ConstructQueryContext* ctx) {
+ParsedQuery Visitor::visit(Parser::ConstructQueryContext* ctx) {
   if (!ctx->datasetClause().empty()) {
     reportError(ctx->datasetClause(0),
                 "QLever currently doesn't support FROM clauses");
   }
   ParsedQuery query;
   if (ctx->constructTemplate()) {
-    query._clause = visitTypesafe(ctx->constructTemplate())
+    query._clause = visit(ctx->constructTemplate())
                         .value_or(parsedQuery::ConstructClause{});
-    auto [pattern, visibleVariables] = visitTypesafe(ctx->whereClause());
+    auto [pattern, visibleVariables] = visit(ctx->whereClause());
     query._rootGraphPattern = std::move(pattern);
     query.registerVariablesVisibleInQueryBody(visibleVariables);
   } else {
     query._clause = parsedQuery::ConstructClause{
         visitOptional(ctx->triplesTemplate()).value_or(Triples{})};
   }
-  query.addSolutionModifiers(visitTypesafe(ctx->solutionModifier()));
+  query.addSolutionModifiers(visit(ctx->solutionModifier()));
 
   return query;
 }
 
 // ____________________________________________________________________________________
-Variable Visitor::visitTypesafe(Parser::VarContext* ctx) {
+ParsedQuery Visitor::visit(Parser::DescribeQueryContext* ctx) {
+  reportError(ctx, "DESCRIBE Queries are not supported.");
+}
+
+// ____________________________________________________________________________________
+ParsedQuery Visitor::visit(Parser::AskQueryContext* ctx) {
+  reportError(ctx, "ASK Queries are not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::DatasetClauseContext* ctx) {
+  reportError(ctx, "FROM is not supported.");
```

Comment:
exactly the same messages as above (FROM clauses are).
And maybe this is an abstraction
```
reportNotSupported(ctx, "FROM clauses are");
```
(or something similar).